src/solver/attract.jl

```diff
@@ -0,0 +1,305 @@
+    # or do we do this traversing outside? in the mother func attract?
+    b = false
+    if oper === (+)
+        for arg in args
+            b |= detect_doubleangle1(arg, var)
+        end
+    elseif oper === (*)
+        b |= detect_doubleangle1(u_lhs, var)
+    end
+    return b
+end
+
+
+"""
+    detect_addlog(lhs, var)
+
+Traverses expression arg by arg and if two logs of
+degree 1 are found it returns true if both of them
+have n_func_occ of 1, so they're solvable after 
+getting attracted. Returns true if found, 
+false if not.
+
+# Arguments
+- lhs: A Num expression.
+- var: Num and should satisfy is_singleton
+
+# Examples
+```jldoctest
+julia> detect_addlogs(log(x+1) + log(x-1), x)
+true
+```
+"""
+function detect_addlogs(lhs, var)
+    u_lhs = unwrap(lhs)
+    args = arguments(u_lhs)
+    oper = operation(u_lhs)
+    !isequal(oper, (+)) && return false
+    
+    found = [false, false]
+    c = 1
+    for arg in args
+        !iscall(arg) && continue
+        isequal(n_occurrences(arg, var), 0) && continue
+
+        # arg has target var and iscall
+        if arg_contains_log(arg, var) && c < 3
+            found[c] = true
+            c += 1
+        end
+    end
+
+    return all(found)
+end
+
+"""
+    detect_exponential(lhs, var)
+
+Traverses an expression and attempts to match the
+expression to ``a*b^f(x) + c*d^g(x)``. Hence why it
+expects only 2 arguments in the input expression.
+Returns true if found, false if not.
+
+# Arguments
+- lhs: A Num expression.
+- var: Num and should satisfy is_singleton
+
+# Examples
+```jldoctest
+julia> detect_exponential(2*3^(x+1) + 5^(x^2 + 3), x)
+true
+```
+"""
+function detect_exponential(lhs, var)
+    lhs = unwrap(lhs)
+    !iscall(lhs) && return false
+    args = arguments(lhs)
+    oper = operation(lhs)
+    !isequal(oper, (+)) && return false
+    length(args) != 2 && return false
+
+    found = [false, false]
+    c = 1
+    for arg in args
+        n_occurrences(arg, var) == 0 && continue
+        !iscall(arg) && continue
+
+        oper = operation(arg)
+        if isequal(oper, ^)
+            args_arg = arguments(arg)
+            n_occurrences(args_arg[1], var) != 0 && continue
+            !check_poly_inunivar(args_arg[2], var) && continue
+
+            found[c] = true
+            c += 1
+
+        elseif isequal(oper, *) && find_logandexpon(arg, var, ^, 2)
+            found[c] = true
+            c += 1
+        end
+    end
+
+    return all(found) 
+end
+
+"""
+    attract_logs(lhs, var)
+
+Rewrites any 2 logs in lhs where the log contains the
+input variable var. so ``log(x) + log(y)`` is not attracted.
+When attracted, it returns an slog.
+
+# Arguments
+- lhs: A Num expression.
+- var: Num and should satisfy is_singleton
+
+# Examples
+```jldoctest
+julia> RootFinding.attract_logs(log(x+1)+log(x-1), x)
+Main.RootFinding.slog(-1 + x^2)
+```
+"""
+function attract_logs(lhs, var)
+    contains_var(arg) = occursin(string(var), string(arg))
```

Comment:
I guess we would need to traverse `arg` to detect the presense of `var`. The current version might break if there are two variables with similar names.

Reply:
true.

Reply:
done